### PR TITLE
Backport of #30306, #31591, and  #31973: NanoGen support in 10_6_X

### PR DESCRIPTION
**Pull request by kdlong**
#### PR description:

Backport of #30306, #31591, and  #31973 in order to have NanoGen support in the release version where UL GEN and NANOAOD are being produced

#### PR validation:

NanoGen workflows run, also tested some private samples.

### CODE REVIEW

**Comment by kpedro88** [comment]
please test

**Comment by chayanit** [comment]
+1

**Comment by jfernan2** [comment]
+1

**Comment by santocch** [comment]
+1

File: PhysicsTools/NanoAOD/python/nanogenDQM_cff.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        GenMET = nanoDQM.vplots.GenMET,
+        GenPart = nanoDQM.vplots.GenPart, 
+        GenVisTau = nanoDQM.vplots.GenVisTau,
+        LHEPart = nanoDQM.vplots.LHEPart,
+        LHEScaleWeight = nanoDQM.vplots.LHEScaleWeight,
+        LHEPdfWeight = nanoDQM.vplots.LHEPdfWeight,
+        PSWeight = nanoDQM.vplots.PSWeight,
+    )
+)
+
```
Comment (mariadalfonso) [inline PhysicsTools/NanoAOD/python/nanogenDQM_cff.py:22]: in master you were importing 
from DQMServices.Core.DQMQualityTester import DQMQualityTester
any reason why not doing this here ?

File: PhysicsTools/NanoAOD/python/nanogen_cff.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    particleLevelTables+
+    metMCTable+
+    genWeightsTable+
+    lheInfoTable
+)
+
+def nanoGenCommonCustomize(process):
+    process.rivetMetTable.extension = False
+    process.lheInfoTable.storeLHEParticles = True
+    process.lheInfoTable.precision = 14
```
Comment (mariadalfonso) [inline PhysicsTools/NanoAOD/python/nanogen_cff.py:44]: you should add the process.genWeightsTable.keepAllPSWeights = True
this was done in master in cms-sw#31276

**Review by mariadalfonso — COMMENTED** [review]
backport is missing modifications done here
https://github.com/cms-sw/cmssw/pull/30306/files#diff-143388ce2c9c75d53018257e6771622d668ba88c1edb3dd8ddbf423656b06e86

File: DQMOffline/Configuration/python/autoDQM.py [hunk 3]
```diff
@@ -100,6 +100,15 @@
                            'PostDQMOffline',
                            'DQMHarvestNanoAOD'],
 
+            'nanogenDQM': ['DQMOfflineNanoGen',
+                           'PostDQMOffline',
+                           'DQMHarvestNanoAOD'],
+
+            'pfDQM': ['DQMOfflinePF+DQMOfflinePFExtended',
+                      'PostDQMOffline',
+                      'DQMHarvestPF'],
```
[the remark this answers is not in the document]
Comment (kdlong) [reply DQMOffline/Configuration/python/autoDQM.py:109]: Oops, thanks, let me remove this.

File: PhysicsTools/NanoAOD/python/nanogenDQM_cff.py (hunk 1 above)
Comment (kdlong) [reply PhysicsTools/NanoAOD/python/nanogenDQM_cff.py:22]: Seems that this refactor happened after 10_6_X, so I reverted back to the older setup here. Note that this follows nanoDQM.py in 10_6_X.

File: PhysicsTools/NanoAOD/python/nanogen_cff.py (hunk 2 above)
Comment (kdlong) [reply PhysicsTools/NanoAOD/python/nanogen_cff.py:44]: Good point, thanks

**Comment by SiewYan** [comment]
+1

**Comment by kdlong** [comment]
@mariadalfonso I think all your concerns are addressed now, to make the backport fully consistent with master. Also implemented a bug fix in the precision setting pointed out by @sroychow, PR to master is #31973

**Comment by mariadalfonso** [comment]
please test workflow 546,547,548

**Comment by mariadalfonso** [comment]
> @mariadalfonso I think all your concerns are addressed now, to make the backport fully consistent with master. Also implemented a bug fix in the precision setting pointed out by @sroychow, PR to master is #31973

please add the 31973 in the title and PR description as well

**Comment by mariadalfonso** [comment]
please test workflow 546,547,548

**Comment by mariadalfonso** [comment]
+xpog
nanoGEN : completed backport to 10_6 inline with changes in master

**Comment by chayanit** [comment]
+1

**Comment by SiewYan** [comment]
+1

**Comment by jfernan2** [comment]
+1

**Comment by mariadalfonso** [comment]
@kpedro88 any request on this PR ?

**Comment by kpedro88** [comment]
+upgrade

**Comment by santocch** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**